server/http_tunnel.go

```diff
@@ -31,22 +32,36 @@ func StartHTTPSTunnel(addr, externalProxy, certFile, KeyFile string) {
 		}}
 		proxy.ConnectDial = proxy.NewConnectDialToProxy(externalProxy)
 	}
+
+	proxy.OnRequest().HandleConnectFunc(func(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
```

Comment:
why did you need to add this? If this is just for debugging purposes, would that be okay to do this under the next func()? `proxy.OnRequest().DoFunc()`?

I could see there would be duplicate debug info when there is a connect request coming in. Some thing like:
```
$ HTTP Tunnel: got CONNECT request $HOST
$ HTTP Tunnel: got request $URL
```

Reply:
Connect and other requests go through different code paths.  In my testing, I never got a request that generated both outputs.  For more info, see this part of the README: https://github.com/elazarl/goproxy/blob/master/README.md#type-of-handlers-for-manipulating-connectreqresp-behavior .

Reply:
Sounds good.